From ca8bc051a92fb351da8d2bcff6256d09c891040a Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 25 Mar 2014 13:40:23 +0200
Subject: [PATCH 1511/1825] fix: alp,a375: eth complex: fix shutdown PHY
 sequence (set SMI source to CPU)

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 1e656321fa0a1852a4ec83c4d871a7d56158d26c

	- Update the SMI source of internal Phys-1/2 to CPU instead of Switch,
	  to access the Phys and Shutdown them.
	- Disable the external SMI when not using external PHYs (when MAC0 <-> GE-PHY0 & MAC1 <-> GE-PHY3)

Change-Id: I6c70b1c92d91f12f42a50abd32c4bba344826b46
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/6644
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.c    |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.c
index 5202c02..e76957d 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEthCompLib.c
@@ -484,9 +484,17 @@ MV_STATUS mvEthComplexInit(MV_U32 ethCompConfig)
 
 	if (c & MV_ETHCOMP_SW_P1_2_GE_PHY_P1)
 		mvEthComplexSwPortToGbePhy(1, 1);
+	else
+		/* Set SMI sourse of PHY-1 to CPU (and not Switch)
+		 to access and shutdown the PHY */
+		mvEthComplexGphyPortSmiSrcSet(1, 0);
 
 	if (c & MV_ETHCOMP_SW_P2_2_GE_PHY_P2)
 		mvEthComplexSwPortToGbePhy(2, 2);
+	else
+		/* Set SMI sourse of PHY-1 to CPU (and not Switch)
+		 to access and shutdown the PHY */
+		mvEthComplexGphyPortSmiSrcSet(2, 0);
 
 	if (c & MV_ETHCOMP_SW_P3_2_GE_PHY_P3)
 		mvEthComplexSwPortToGbePhy(3, 3);
-- 
1.7.5.4

